features/attach_invalid_token.feature

```diff
@@ -0,0 +1,15 @@
+Feature: Command behaviour when trying to attach a machine to an Ubuntu
+         Advantage subscription using an invalid token
+
+    Scenario: Attach command in a trusty lxd container
+       Given a trusty lxd container with ubuntu-advantage-tools installed
+        When I attach "<contract_token>" with sudo
```

Comment:
weird, because I was printing it to make sure I had the token and then `
`['sudo', 'ua', 'attach', '0dieQQV2IFx2GvOLN6dsZJjOuIYEK']` in this case I didn't provide it via CLI. so it is a random token, but Ill double check it

Reply:
sorry that the content was unclear. I added the raise RuntimeError(token) to the first line inside your when_i_attach_token() method, before you performed the reset of `token = context.config.contract_token`

The issue I was trying to suggest here wasn't that you didn't override that token value, my point was instead that when_i_attach_token() method shouldn't completely ignore it's template input variable 'token' param in the first place. That `token` method param should somehow be used to "lookup" various token values off of context.config.

As you currently have written when_i_attach_token() , you have hard-coded this method to always use `context.config.contract_token` it doesn't matter if our feature file has a

 When I attach "<contract_token>" with sudo
or 
When I attach "some_valid_token_string" with sudo

both Scenario When statements will only be exercise ua attach <context.config.contract_token>